---
title: "NANOSED - Adsorption of metallic nanoparticles to estuarine sediments: what implication for denitrification?"
author: "Authors: Nanosed Team"
date: "Last update 10/03/2020"
output: html_document
---
Data analysis of the work entitled "Copper nanoparticles prompt the activity of estuarine denitrifying bacterial communities at relevant environmental concentrations" <br> DOI:<to fill>

```{r setup, include=FALSE}
knitr::opts_chunk$set(
  echo = TRUE, 
  warning=FALSE
)
```

```{r Libraries, message=FALSE, warning=FALSE, paged.print=FALSE}

# Libraries
library("tidyverse")     # data and plots    
library("vegan")         # ecological multivariate analyses
library("dabestr")       # test significance no p-value based
library("envalysis")     # environmental analysis
library("scales")        # map data to aesthetics
library("cowplot")       # arranging plots
```

```{r Seed}
set.seed(57)
```

```{r Data}
# Import data
data <- read.csv("Copper_nanoparticle_estuary_microcosms.csv")
```
#### Changes in microcosms due to salinity
```{r Salinity microcosms, dpi = 96}

# Select data for salinity comparison
data_sal <- data %>% 
  select(EXP, Time, Conc, Size, Salinity, O2, pH, OM, 
         NO3wat, NO2wat, NH4wat, NO3sed, NO2sed, NH4sed,
         Cuenv, Feenv) %>% 
  filter(EXP == "EXP1" | EXP == "EXP2")

# Transform data with z-score
data_zscores<- scale(data_sal[,c(6:16)],center = T, scale = T) %>% as.data.frame()

# Create factors in metadata
Metadata <- data_sal %>% 
  select(Time, Conc, Salinity)
Metadata$Time <- factor(Metadata$Time)
Metadata$Conc <- factor(Metadata$Conc)

# Calculate distance - Euclidean 
dis.eucli <- vegdist(data_zscores, method="euclidean", na.rm=TRUE)

# PCoA
pcoa.ord <- cmdscale(dis.eucli, eig = T)

# Create a dataframe to plot the PCoA 
PCoA.ord <-data.frame(PCoA1 = pcoa.ord$points[,1], PCoA2 = pcoa.ord$points[,2])
PCoA.ord.df<- merge(PCoA.ord , Metadata, by="row.names", sort = F)
PCoA.ord.df <- PCoA.ord.df[,-1]

# How much % of the variance is explained by the axes?
round(pcoa.ord$eig*100/sum(pcoa.ord$eig),1)

# Plot PCoA
ggplot(data = PCoA.ord.df, aes(PCoA1, PCoA2)) + 
  geom_point(aes(color = Time, shape = Conc, size = 1)) +
  labs( y="PCoA2 (19.6%)", x="PCoA1 (22.5%)") +
  facet_wrap(Salinity~.) +
  scale_color_manual(name="Time (h)",
                     labels = c("1","4","7","24","48","144"),
                     values =c ("#E64B35FF","#4DBBD5FF","#3C5488FF","#00A087FF","#F39B7FFF","#91D1C2FF")) +
  scale_shape_manual(name="Conc (µg/g)",
                   labels = c("0","0.01","0.1","1"),
                   values = c(16,17,15,3)) +
  guides(size = F, 
         shape = guide_legend(order=1, override.aes = list(size=3)), 
         color = guide_legend(order=2, override.aes = list(size=3))) +
  theme_publish() +
  theme(
    axis.title.x = element_text(size=16),
    axis.text.x  = element_text(size=16),
    axis.title.y = element_text(size=16),
    axis.text.y  = element_text(size=16),
    legend.title = element_text(size=16),
    legend.text = element_text(size=16),
    legend.position = "right",
    legend.box = "vertical") +
  ggtitle("Microcosms with Cu NPs < 50 nm - effect of salinity")

# Check for differences between Groups - PERMANOVA
adonis2(dis.eucli ~ Salinity*Time, data = Metadata, by ="margin")
adonis2(dis.eucli ~ Salinity*Conc, data = Metadata, by ="margin")

# Are differences in PERMANOVA due to the dispersion of the results?
beta_dispersion_data <- betadisper(dis.eucli, Metadata$Time)
permutest(beta_dispersion_data)
```
Principal coordinates analysis shows that for both salinities the microcosms displayed the same behaviour throughout time. PERMANOVA showed that differences between Cu concentrations are not significant, but between different time points are significant. Since the dispersion of the data was p = 0.01 the results of the PERMANOVA were interpreted as reflecting different dispersions over time. <br> To further investigate the effect of salinity, linear regressions of inorganic N compounds (in water and sediment), metal concentration in sediment (environmentally available), organic matter in sediment (OM), water dissolved O2 and pH were analysed over time. 

```{r Microcosms salinity time, dpi = 96}

# Lm model - linear response

# Create a data frame for each EXP
data_lm_EXP1 <- data_sal %>% filter(EXP == "EXP1") 
data_lm_EXP2 <- data_sal %>% filter(EXP == "EXP2")

# EXP1
x1 <- data_lm_EXP1 %>% select(Time)
y1 <- data_lm_EXP1  %>%
  select(O2, pH,  OM, NO3wat, NO2wat, NH4wat, NO3sed, NO2sed, NH4sed, Cuenv, Feenv)

out_lm1 <- data.frame(NULL)              
for (i in 1:length(y1)) {
  m1 <- summary(lm(y1[,i] ~ x1[,1]))       
  out_lm1[i, 1] <- names(y1)[i]           
  out_lm1[i, 2] <- m1$coefficients[1,1]  
  out_lm1[i, 3] <- m1$coefficients[2,1]   
  out_lm1[i, 4] <- m1$coefficients[2,4]   
  }
names(out_lm1) <- c("y.variable", "intercept", "slope", "p.value")
out_lm1

# EXP2
x2 <- data_lm_EXP2 %>% select(Time)
y2 <- data_lm_EXP2  %>%
  select(O2, pH, OM, NO3wat, NO2wat, NH4wat, NO3sed, NO2sed, NH4sed, Cuenv, Feenv)

out_lm2 <- data.frame(NULL)              
for (i in 1:length(y2)) {
  m2 <- summary(lm(y2[,i] ~ x2[,1]))       
  out_lm2[i, 1] <- names(y2)[i]           
  out_lm2[i, 2] <- m2$coefficients[1,1]  
  out_lm2[i, 3] <- m2$coefficients[2,1]   
  out_lm2[i, 4] <- m2$coefficients[2,4]   
  }
names(out_lm2) <- c("y.variable", "intercept", "slope", "p.value")
out_lm2

# T-test to compare slopes of two regressions 

#####
fit1_ph <- lm(pH~Time, data_lm_EXP1)
s1_ph <- summary(fit1_ph)$coefficients
fit2_ph <- lm(pH~Time, data_lm_EXP2)
s2_ph <- summary(fit2_ph)$coefficients
#
db_ph <- (s2_ph[2,1]-s1_ph[2,1])
sd_ph <- sqrt(s2_ph[2,2]^2+s1_ph[2,2]^2)
df_ph <- (fit1_ph$df.residual+fit2_ph$df.residual)
td_ph <- db_ph/sd_ph
t_test_ph <- 2*pt(-abs(td_ph), df_ph)
#####
fit1_no3 <- lm(NO3wat~Time, data_lm_EXP1)
s1_no3 <- summary(fit1_no3)$coefficients
fit2_no3 <- lm(NO3wat~Time, data_lm_EXP2)
s2_no3 <- summary(fit2_no3)$coefficients
#
db_no3 <- (s2_no3[2,1]-s1_no3[2,1])
sd_no3 <- sqrt(s2_no3[2,2]^2+s1_no3[2,2]^2)
df_no3 <- (fit1_no3$df.residual+fit2_no3$df.residual)
td_no3 <- db_no3/sd_no3
t_test_no3 <- 2*pt(-abs(td_no3), df_no3)

#####
out_fit <- data.frame(NULL)  
  out_fit[1, 1] <- t_test_ph           
  out_fit[1, 2] <- t_test_no3 
names(out_fit) <- c("pH", "NO3 water")
out_fit
```
For both salinities pH and N-NO3 concentration in water increased over time (p < 0.001). A t-test compared the slopes obtained for each salinity and no differences could be found, showing similar increases over time.

#### Changes in microcosms with different NPs sizes
```{r Microcosms NPs addition, dpi = 96}

data_np <- data %>% 
  select(EXP, Time, Conc, Size, Salinity, O2, pH, OM, 
         NO3wat, NO2wat, NH4wat, NO3sed, NO2sed, NH4sed,
         Cuenv, Feenv, Cuexc, Feexc, DAPI) %>%
  filter(EXP == "EXP50" | EXP == "EXP150")

# Normalize data with z-score
data_np_zscores <- scale(data_np[,c(6:18)], center = T, scale = T) %>% as.data.frame()

# Create factors in metadata
Metadata_np <- data_np %>% 
  select(Time, Conc, Size)
Metadata_np$Time <- factor(Metadata_np$Time)
Metadata_np$Conc <- factor(Metadata_np$Conc)

# Calculate distances - Euclidean 
dis.eucli.np <- vegdist(data_np_zscores, method="euclidean", na.rm=TRUE)

# PCoA 
pcoa.ord.np <- cmdscale(dis.eucli.np, eig = T)

# Create a dataframe to plot the PCoA 

# Create a dataframe to plot the PCoA 
PCoA.ord.np <-data.frame(PCoA1 = pcoa.ord.np$points[,1], PCoA2 = pcoa.ord.np$points[,2])
PCoA.ord.np.df<- merge(PCoA.ord.np, Metadata_np, by="row.names", sort = F)
PCoA.ord.np.df <- PCoA.ord.np.df[,-1]

# How much % of the variance is explained by the axes?
round(pcoa.ord.np$eig*100/sum(pcoa.ord.np$eig),1)

# Order the NPs sizes
PCoA.ord.np.df$Size = factor(PCoA.ord.np.df$Size, levels=c("50nm","150nm"))

# Plot PCoA
ggplot(data = PCoA.ord.np.df, aes(PCoA1, PCoA2)) + 
  geom_point(aes(color = Size, shape = Time, size = 1)) +
  stat_ellipse(aes(colour = Size, group = Size)) +
  labs( y="PCoA2 (16.1%)", x="PCoA1 (28.4%)") +
  scale_color_manual(name="Size",
                     labels = c("50nm","150nm"),
                     values = c("#FBB4AE","#B3CDE3")) +
  scale_shape_manual(name="Time (h)",
                     labels = c("1","4","7","24","48","144"),
                     values =c (15,16,17,3,4,5)) +
  guides(size = F, 
         shape = guide_legend(order=2, override.aes = list(size=3)), 
         color = guide_legend(order=1, override.aes = list(size=2))) +
  theme_publish() +
  theme(
    axis.title.x = element_text(size=16),
    axis.text.x  = element_text(size=16),
    axis.title.y = element_text(size=16),
    axis.text.y  = element_text(size=16),
    legend.title = element_text(size=16),
    legend.text = element_text(size=16),
    legend.position = "right",
    legend.box = "vertical") +
  ggtitle("Microcosms with Cu NPs < 50 nm and < 150 nm")

# Check for differences between Groups - PERMANOVA
adonis2(dis.eucli.np ~ Size * Conc, data = Metadata_np)
adonis2(dis.eucli.np ~ Size * Time, data = Metadata_np)

# Are differences in PERMANOVA due to the dispersion of the results?
beta_dispersion_np_size <- betadisper(dis.eucli.np, Metadata_np$Size)
permutest(beta_dispersion_np_size)

beta_dispersion_np_time <- betadisper(dis.eucli.np, Metadata_np$Time)
permutest(beta_dispersion_np_time)
```
Principal coordinates analysis shows that for different sizes the microcosms displayed a different behaviour throughout time. PERMANOVA showed size and time as significant. Since the dispersions were not significant the results of the PERMANOVA were interpreted as not reflecting different dispersions. <br> To further investigate the effect of time, linear regressions of inorganic N compounds (in water and sediment), metal concentration in sediment (environmentally available and exchangeable), organic matter in sediment (OM), water dissolved O2 and pH were analysed over time. 

```{r Microcosms NPs addition time, dpi = 96}

# Lm model - linear response

# Create a data frame for each EXP
data_lm_EXP50 <- data_np %>% filter(EXP == "EXP50") 
data_lm_EXP150 <- data_np %>% filter(EXP == "EXP150")

# EXP50
x1 <- data_lm_EXP50 %>% select(Time)
y1 <- data_lm_EXP50  %>%
  select(O2, pH, OM, NO3wat, NO2wat, NH4wat, NO3sed, NO2sed, NH4sed, Cuenv, Feenv, Cuexc, Feexc, DAPI)

out_lm50 <- data.frame(NULL)              
for (i in 1:length(y1)) {
  m1 <- summary(lm(y1[,i] ~ x1[,1]))       
  out_lm50[i, 1] <- names(y1)[i]           
  out_lm50[i, 2] <- m1$coefficients[1,1]  
  out_lm50[i, 3] <- m1$coefficients[2,1]   
  out_lm50[i, 4] <- m1$coefficients[2,4]   
  }
names(out_lm50) <- c("y.variable", "intercept", "slope", "p.value")
out_lm50

# EXP150
x2 <- data_lm_EXP150 %>% select(Time)
y2 <- data_lm_EXP150 %>%
  select(O2, pH, OM, NO3wat, NO2wat, NH4wat, NO3sed, NO2sed, NH4sed, Cuenv, Feenv, Cuexc, Feexc, DAPI)

out_lm150 <- data.frame(NULL)              
for (i in 1:length(y2)) {
  m2 <- summary(lm(y2[,i] ~ x2[,1]))       
  out_lm150[i, 1] <- names(y2)[i]           
  out_lm150[i, 2] <- m2$coefficients[1,1]  
  out_lm150[i, 3] <- m2$coefficients[2,1]   
  out_lm150[i, 4] <- m2$coefficients[2,4]   
  }
names(out_lm150) <- c("y.variable", "intercept", "slope", "p.value")
out_lm150

# T-test to compare slopes of two regressions 

#####
fit1_ph <- lm(pH~Time, data_lm_EXP50)
s1_ph <- summary(fit1_ph)$coefficients
fit2_ph <- lm(pH~Time, data_lm_EXP150)
s2_ph <- summary(fit2_ph)$coefficients
#
db_ph <- (s2_ph[2,1]-s1_ph[2,1])
sd_ph <- sqrt(s2_ph[2,2]^2+s1_ph[2,2]^2)
df_ph <- (fit1_ph$df.residual+fit2_ph$df.residual)
td_ph <- db_ph/sd_ph
t_test_ph <- 2*pt(-abs(td_ph), df_ph)
#####
fit1_no3 <- lm(NO3wat~Time, data_lm_EXP50)
s1_no3 <- summary(fit1_no3)$coefficients
fit2_no3 <- lm(NO3wat~Time, data_lm_EXP150)
s2_no3 <- summary(fit2_no3)$coefficients
#
db_no3 <- (s2_no3[2,1]-s1_no3[2,1])
sd_no3 <- sqrt(s2_no3[2,2]^2+s1_no3[2,2]^2)
df_no3 <- (fit1_no3$df.residual+fit2_no3$df.residual)
td_no3 <- db_no3/sd_no3
t_test_no3 <- 2*pt(-abs(td_no3), df_no3)
#####
fit1_no2 <- lm(NO2sed~Time, data_lm_EXP50)
s1_no2 <- summary(fit1_no2)$coefficients
fit2_no2 <- lm(NO2sed~Time, data_lm_EXP150)
s2_no2 <- summary(fit2_no2)$coefficients
#
db_no2 <- (s2_no2[2,1]-s1_no2[2,1])
sd_no2 <- sqrt(s2_no2[2,2]^2+s1_no2[2,2]^2)
df_no2 <- (fit1_no2$df.residual+fit2_no2$df.residual)
td_no2 <- db_no2/sd_no2
t_test_no2 <- 2*pt(-abs(td_no2), df_no2)
#####
fit1_fe <- lm(Feexc~Time, data_lm_EXP50)
s1_fe <- summary(fit1_fe)$coefficients
fit2_fe <- lm(Feexc~Time, data_lm_EXP150)
s2_fe <- summary(fit2_fe)$coefficients
#
db_fe <- (s2_fe[2,1]-s1_fe[2,1])
sd_fe <- sqrt(s2_fe[2,2]^2+s1_fe[2,2]^2)
df_fe <- (fit1_fe$df.residual+fit2_fe$df.residual)
td_fe <- db_fe/sd_fe
t_test_fe <- 2*pt(-abs(td_fe), df_fe)

#####
out_fit <- data.frame(NULL)  
  out_fit[1, 1] <- t_test_ph           
  out_fit[1, 2] <- t_test_no3 
  out_fit[1, 3] <- t_test_no2
  out_fit[1, 4] <- t_test_fe
  names(out_fit) <- c("pH", "NO3 water", "NO2 sed","Fe exchangeable")
out_fit
```
pH, N-NO3 concentration in water, N-NO2 concentration in sediment and Fe exchangeable fraction in sediment increase over time (p < 0.001) at least for one size of Cu NPs. A t-test compared the slopes of each parameter in microcosms with both Cu NPs sizes and differences could only be found for N-NO3 concentration in water, which increased more on microcosms exposed to Cu NPs < 50 nm.

#### Changes in number of cells
```{r Microcosms NPs addition cell, dpi = 96}

# Create factors in data_np
data_np$Size <- factor(data_np$Size, levels=c("50nm","150nm"))
data_np$Conc <- factor(data_np$Conc)

# Evaluate increases in cell number

# Test for data normality with Anova & Shapiro-Wilk test (if not normal use Kruskal-wallis)
dapi.aov <- aov(DAPI ~ Conc, data = data_np)
dapi.aov_residuals <- residuals(object = dapi.aov)
shapiro.test(x = dapi.aov_residuals)

# Kruskal-wallis
kruskal.test(DAPI ~ Conc, data = data_np)

# If significant compare control without NPs with microcosms with NPs - Wilcoxon signed-rank test
pairwise.wilcox.test(data_np$DAPI, data_np$Conc, p.adjust.method = "BH")

# Estimation plot

DAPI_best <- dabest(data_np, Conc, DAPI,
                             idx = c("0", "0.01", "0.1", "1"),
                             paired = FALSE)

plot(DAPI_best, color.column = Size,
     rawplot.markersize = 3,
     rawplot.ylabel = "DAPI-stained cells \n (number/g sediment)",
     effsize.ylabel = "Unpaired \n mean difference",
     axes.title.fontsize = 10,
     palette = "Pastel1")
```
Kruskal-wallis test shows increases in cell number are significant. An estimation plot shows the mean difference in cell number is higher in microcosms exposed to 0.01 and 1 µg/g Cu NPs < 50 nm.

#### Changes in metal content
```{r Microcosms NPs addition metal, dpi = 96}

# Evaluate increases in metal concentration

# Test for data normality with Anova & Shapiro-Wilk test
cuenv.aov <- aov(Cuenv ~ Conc, data = data_np)
cuenv.aov_residuals <- residuals(object = cuenv.aov)
shapiro.test(x = cuenv.aov_residuals)

# Kruskal-wallis test
kruskal.test(Cuenv ~ Conc, data = data_np)

# Test for data normality with Anova & Shapiro-Wilk test
cuexc.aov <- aov(Cuexc ~ Conc, data = data_np)
cuexc.aov_residuals <- residuals(object = cuexc.aov)
shapiro.test(x = cuexc.aov_residuals)

# Kruskal-wallis test
kruskal.test(Cuexc ~ Conc, data = data_np)
pairwise.wilcox.test(data_np$Cuexc, data_np$Conc, p.adjust.method = "BH")

# Test for data normality with Anova & Shapiro-Wilk test
feenv.aov <- aov(Feenv ~ Conc, data = data_np)
feenv.aov_residuals <- residuals(object = feenv.aov)
shapiro.test(x = feenv.aov_residuals)

# Kruskal-wallis test (data is not normal)
kruskal.test(Feenv ~ Conc, data = data_np)

# Test for data normality with Anova & Shapiro-Wilk test
feexc.aov <- aov(Feexc ~ Conc, data = data_np)
feexc.aov_residuals <- residuals(object = feexc.aov)
shapiro.test(x = feexc.aov_residuals)

# Kruskal-wallis test
kruskal.test(Feexc ~ Conc, data = data_np)

# Estimation plots

Cu_env <- dabest(data_np, Conc, Cuenv,
                             idx = c("0", "0.01", "0.1", "1"),
                             paired = FALSE)
Cu_exc <- dabest(data_np, Conc, Cuexc,
                             idx = c("0", "0.01", "0.1", "1"),
                             paired = FALSE)
Fe_env <- dabest(data_np, Conc, Feenv,
                             idx = c("0", "0.01", "0.1", "1"),
                             paired = FALSE)
Fe_exc <- dabest(data_np, Conc, Feexc,
                             idx = c("0", "0.01", "0.1", "1"),
                             paired = FALSE)

p1<- plot(Cu_env, color.column = Size,
     rawplot.markersize = 2,
     rawplot.ylabel = "Cu environmental \n (µg/g sediment)",
     effsize.ylabel = "Unpaired \n mean difference",
     axes.title.fontsize = 10,
     palette = "Pastel1")
p2<- plot(Cu_exc, color.column = Size,
     rawplot.markersize = 2,
     rawplot.ylabel = "Cu exchangeable \n (µg/g sediment)",
     effsize.ylabel = "Unpaired \n mean difference",
     axes.title.fontsize = 10,
     palette = "Pastel1") 
p3<- plot(Fe_env, color.column = Size,
     rawplot.markersize = 2,
     rawplot.ylabel = "Fe environmental \n (mg/g sediment)",
     effsize.ylabel = "Unpaired \n mean difference",
     axes.title.fontsize = 10,
     palette = "Pastel1")
p4<- plot(Fe_exc, color.column = Size,
     rawplot.markersize = 2,
     rawplot.ylabel = "Fe exchangeable \n (mg/g sediment)",
     effsize.ylabel = "Unpaired \n mean difference",
     axes.title.fontsize = 10,
     palette = "Pastel1")

plot_grid1 <- plot_grid(p1, p2, p3, p4, 
          ncol=2, nrow=2)
plot_grid1

# Normalise Cu concentration

data_np <- transform(data_np, ratiocuenv = Cuenv / Feenv)

ratio_Cu <- dabest(data_np, Conc, ratiocuenv,
                             idx = c("0", "0.01", "0.1", "1"),
                             paired = FALSE)
plot(ratio_Cu , color.column = Size,
     rawplot.markersize = 3,
      rawplot.ylabel = "Cu/Fe environmental \n fraction",
     effsize.ylabel = "Unpaired \n mean difference",
     axes.title.fontsize = 10,
     palette = "Pastel1")

```
Kruskal-wallis test shows increases in Cu exchangeable fraction slightly significant. An estimation plot shows the mean difference in Cu exchangeable concentration is higher in microcosms exposed to 1 µg/g Cu NPs < 50 nm.

#### Can 16S be used as reference gene?
```{r Microcosms 16S gene expression, dpi = 96}

# Import data 
data_reference <- data %>% 
  select(EXP, Time, Foldch16S)  %>%
  filter(EXP == "EXP50" | EXP == "EXP150")

ggplotRegression <- function (fit) {

    ggplot(fit$model, aes_string(x = names(fit$model)[2], y = names(fit$model)[1])) + 
      geom_point(color = "indianred") +
    ggtitle("Linear regression of fold change in 16S expression over time") +
    theme_publish() +
    ylab("Log2 Fold change") +
    xlab("Time (h)") +
      stat_smooth(method = "lm", color = "black") +
      labs(subtitle = paste("R2 = ", signif(summary(fit)$r.squared, 3),
                         "Intercept =", signif(fit$coef[[1]], 3),
                         "Slope =", signif(fit$coef[[2]], 3),
                         "P =", signif(summary(fit)$coef[2,4], 3)))
    }

ggplotRegression(lm(Foldch16S ~ Time, data = data_reference))
```
16S expression did not change over time, therefore, it can be used as a reference gene.

#### Changes gene expression
```{r Microcosms gene expression, dpi = 96}

data_gene <- data %>% 
  select(EXP, Time, Conc, Size, nosZ, nirS, nirK) %>%
  filter(EXP == "EXP50" | EXP == "EXP150") %>%
  filter(Conc != "0")

data_gene$Conc <- factor(data_gene$Conc)
data_gene$Time <- factor(data_gene$Time, levels = rev(unique(data_gene$Time)))

conc.labs <- c("0 µg/g","0.01 µg/g", "0.1 µg/g", "1 µg/g")
names(conc.labs) <- c("0","0.01", "0.1", "1")

plot_nirS50 <- ggplot(data_gene %>% filter(EXP=="EXP50"), aes(x = Time, y = nirS, fill = nirS>0)) + 
  geom_bar(stat = "identity", position = "dodge") +  
  facet_grid(~Conc, labeller = labeller(Conc = conc.labs)) +
  scale_fill_manual(values = c("#FDDBC7","#E64B35FF")) +
  ylab(expression(paste("Log"[2], " fold-change in ", italic("nirS"), " transcripts "))) + 
  xlab("Time (h)") + 
  geom_hline(yintercept = 0) + 
  coord_flip() + 
  scale_y_continuous(breaks = seq(-8, 14, by = 4)) +
  guides(fill=F) +
  theme_publish()

plot_nirS150 <- ggplot(data_gene %>% filter(EXP=="EXP150"), aes(x = Time, y = nirS, fill = nirS>0)) + 
  geom_bar(stat = "identity", position = "dodge") + 
  facet_grid(~Conc, labeller = labeller(Conc = conc.labs)) +
  scale_fill_manual(values = c("#FDDBC7","#E64B35FF")) +
  ylab(expression(paste("Log"[2], " (fold-change) in ", italic("nirS"), " transcripts "))) + 
  xlab("Time (h)") + 
  geom_hline(yintercept = 0) + 
  coord_flip() + 
  scale_y_continuous(breaks = seq(-6, 6, by = 2)) +
  guides(fill=F) +
  theme_publish() 

plot_nosZ50 <- ggplot(data_gene %>% filter(EXP=="EXP50"), aes(x = Time, y = nosZ, fill = nosZ>0)) + 
  geom_bar(stat = "identity", position = "dodge") + 
  facet_grid(~Conc, labeller = labeller(Conc = conc.labs)) +
  scale_fill_manual(values = c("#D1E5F0","#4DBBD5FF")) +
  ylab(expression(paste("Log"[2], " (fold-change) in ", italic("nosZ"), " transcripts "))) + 
  xlab("Time (h)") + 
  geom_hline(yintercept = 0) + 
  coord_flip() + 
  scale_y_continuous(breaks = seq(-8, 14, by = 4)) +
  guides(fill=F) +
  theme_publish()

plot_nosZ150 <- ggplot(data_gene %>% filter(EXP=="EXP150"), aes(x = Time, y = nosZ, fill = nosZ>0)) + 
  geom_bar(stat = "identity", position = "dodge") +  
  facet_grid(~Conc, labeller = labeller(Conc = conc.labs)) +
  scale_fill_manual(values = c("#D1E5F0","#4DBBD5FF")) +
  ylab(expression(paste("Log"[2], " (fold-change) in ", italic("nosZ"), " transcripts "))) + 
  xlab("Time (h)") + 
  geom_hline(yintercept = 0) + 
  coord_flip() + 
  scale_y_continuous(breaks = seq(-6, 6, by = 2)) +
  guides(fill=F) +
  theme_publish()

plot_nirK50 <- ggplot(data_gene %>% filter(EXP=="EXP50"), aes(x = Time, y = nirK, fill = nirK>0)) + 
  geom_bar(stat = "identity", position = "dodge") +  
  facet_grid(~Conc, labeller = labeller(Conc = conc.labs)) +
  scale_fill_manual(values = c("#AEFCEE","#00A087FF")) +
  ylab(expression(paste("Log"[2], " (fold-change) in ", italic("nirK"), " transcripts "))) + 
  xlab("Time (h)") + 
  geom_hline(yintercept = 0) + 
  coord_flip() + 
  scale_y_continuous(breaks = seq(-6, 6, by = 2)) +
  guides(fill=F) +
  theme_publish()

plot_grid2 <- plot_grid(plot_nirS50, plot_nirS150,
                        plot_nosZ50, plot_nosZ150, 
                        plot_nirK50,
          ncol=2, nrow=3)
plot_grid2
```

#### Gene copy number
```{r Microcosms gene copy number, fig.height=10, fig.width=12}

gene_copy <- data %>% 
  select(EXP, Time, Conc, copy16, copynos, copynir) %>%
  filter(EXP == "EXP50")

gene_copy$Conc <- factor(gene_copy$Conc)
gene_copy$Time <- factor(gene_copy$Time)

library(wesanderson)
pal <- wes_palette("Zissou1", 10, type = "continuous")

nir50_heatmap <- ggplot(data = gene_copy, 
         mapping = aes(x = Time, y = Conc, fill = copynir)) +
  geom_tile() +
  scale_fill_gradientn(colours = pal, name = "Copy number / \ng sediment", label=scientific) + 
  scale_x_discrete(expand = c(0, 0)) +
  scale_y_discrete(expand = c(0, 0)) + 
  coord_equal() +
  ylab(expression(Cu~NPs~(µg~g^{-1}))) +
  xlab("Time (h)") +
  theme_publish() +
  theme(
    axis.title.x = element_text(size=16),
    axis.text.x  = element_text(size=16),
    axis.title.y = element_text(size=16),
    axis.text.y  = element_text(size=16),
    legend.title = element_text(size=14),
    legend.text = element_text(size=14),
    legend.position = "right") +
  ggtitle("nirS Cu NPs < 50 nm")

nos50_heatmap <- ggplot(data = gene_copy, 
         mapping = aes(x = Time, y = Conc, fill = copynos)) +
  geom_tile() +
  scale_fill_gradientn(colours = pal, name = "Copy number / \ng sediment", label=scientific) +
  scale_x_discrete(expand = c(0, 0)) +
  scale_y_discrete(expand = c(0, 0)) + 
  coord_equal() +
  ylab(expression(Cu~NPs~(µg~g^{-1}))) +
  xlab("Time (h)") +
  theme_publish() +
  theme(
    axis.title.x = element_text(size=16),
    axis.text.x  = element_text(size=16),
    axis.title.y = element_text(size=16),
    axis.text.y  = element_text(size=16),
    legend.title = element_text(size=14),
    legend.text = element_text(size=14),
    legend.position = "right") +
 ggtitle("nosZ Cu NPs < 50 nm")

ref16S50_heatmap<- ggplot(data = gene_copy, 
         mapping = aes(x = Time, y = Conc, fill = copy16)) +
  geom_tile() +
  scale_fill_gradientn(colours = pal) + 
  scale_x_discrete(expand = c(0, 0)) +
  scale_y_discrete(expand = c(0, 0)) + 
  coord_equal() +
  ylab(expression(Cu~NPs~(µg~g^{-1}))) +
  xlab("Time (h)") +
  theme(
    axis.title.x = element_text(size=16),
    axis.text.x  = element_text(size=16),
    axis.title.y = element_text(size=16),
    axis.text.y  = element_text(size=16),
    legend.title = element_text(size=14),
    legend.text = element_text(size=14),
    legend.position = "right") +
   ggtitle("16S Cu NPs < 50 nm")

plot_grid3 <- plot_grid(nir50_heatmap, nos50_heatmap,ref16S50_heatmap,
                        ncol=1, nrow=3)
plot_grid3

# Create data set for Log copy genes divided by Log 16S

gene_copy_nr<- log10(gene_copy %>% select(copy16, copynos, copynir))
gene_copy_nr<- merge(gene_copy %>% select(Time, Conc), gene_copy_nr, by ="row.names", sort = F)

gene_copy_nr <- transform(gene_copy_nr, lognos = copynos / copy16)
gene_copy_nr <- transform(gene_copy_nr, lognir = copynir / copy16)

plot_gene_nos <-ggplot(gene_copy_nr, aes(x = Conc, y = lognos, fill= Conc)) +
  geom_dotplot(binaxis="y", stackdir="center") +
  geom_boxplot() +
  ylab(expression(Log10~italic(nosZ)~"/"~Log10~italic("16S"))) +
  xlab(expression(Cu~NPs~(µg~g^{-1}~sediment))) +
  scale_fill_manual(values = c("#E64B35FF","#4DBBD5FF","#00A087FF","#3C5488FF")) +
  theme_publish() +
  theme(
    axis.title.x = element_text(size=16),
    axis.text.x  = element_text(size=16),
    axis.title.y = element_text(size=16),
    axis.text.y  = element_text(size=16),
    legend.position ="none") +
  ggtitle("nosZ / 16S Cu NPs < 50 nm")


plot_gene_nir <-ggplot(gene_copy_nr, aes(x = Conc, y = lognir, fill= Conc)) +
  geom_dotplot(binaxis="y", stackdir="center") +
  geom_boxplot() +
  ylab(expression(Log10~italic(nirS)~"/"~Log10~italic("16S"))) +
  xlab(expression(Cu~NPs~(µg~g^{-1}~sediment))) +
  scale_fill_manual(values = c("#E64B35FF","#4DBBD5FF","#00A087FF","#3C5488FF")) +
  theme_publish() +
  theme(
    axis.title.x = element_text(size=16),
    axis.text.x  = element_text(size=16),
    axis.title.y = element_text(size=16),
    axis.text.y  = element_text(size=16),
    legend.position ="none") +
   ggtitle("nirS / 16S Cu NPs < 50 nm")

plot_grid4 <- plot_grid(plot_gene_nir, plot_gene_nos,
                        ncol=2, nrow=1)
plot_grid4 

# Are these differences significant?

kruskal.test(lognir ~ Conc, data = gene_copy_nr) 
pairwise.wilcox.test(gene_copy_nr$lognir, gene_copy_nr$Conc, p.adjust.method = "BH")

kruskal.test(lognos ~ Conc, data = gene_copy_nr)
pairwise.wilcox.test(gene_copy_nr$lognos, gene_copy_nr$Conc, p.adjust.method = "BH")
```

#### Gene copy correl
```{r Microcosms gene correl, fig.height=10, fig.width=12}

ggplotRegression2 <- function (fit) {

    ggplot(fit$model, aes_string(x = names(fit$model)[2], y = names(fit$model)[1])) + 
      geom_point(color = "indianred") +
      stat_smooth(method = "lm", color = "indianred") +
      labs(subtitle = paste("R2 =", signif(summary(fit)$r.squared, 3),
                            "P =", signif(summary(fit)$coef[2,4], 3))) +
      ylab(expression(paste("Log"[10], italic("nirS"), " abundance "))) + 
    xlab(expression(paste("Log"[10], italic("nosZ"), " abundance "))) 
  
    }

p10<- ggplotRegression2(lm(copynir ~ copynos, data = gene_copy_nr[gene_copy_nr$Conc=="0",])) +
  ggtitle("0 µg/g") + theme_publish()
p11<- ggplotRegression2(lm(copynir ~ copynos, data = gene_copy_nr[gene_copy_nr$Conc=="0.01",])) +
  ggtitle("0.1 µg/g") + theme_publish()
p12<- ggplotRegression2(lm(copynir ~ copynos, data = gene_copy_nr[gene_copy_nr$Conc=="0.1",])) +
  ggtitle("0.1 µg/g") + theme_publish()
p13<- ggplotRegression2(lm(copynir ~ copynos, data = gene_copy_nr[gene_copy_nr$Conc=="1",])) +
  ggtitle("1 µg/g") + theme_publish()

plot_grid5 <- plot_grid(p10, p11, p12, p13,
                        ncol=2, nrow=2)
plot_grid5
```

#### Denitrification
```{r Microcosms denitrification, fig.height=10, fig.width=12}

dnf_data <- read.csv("Copper_nanoparticle_estuary_denitrification.csv")

dnf_data$Conc <- factor(dnf_data$Conc)

p14<-ggplot(dnf_data, aes(x=Conc, y=DNF, fill= Conc)) + 
  geom_dotplot(binaxis="y", stackdir="center") +
  geom_boxplot() +
  ylab(expression(nmol~N~~g^{-1}~h^{-1})) +
  xlab(expression(Cu~NPs~(µg~g^{-1}~sediment))) +
  scale_fill_manual(values = c("#E64B35FF","#4DBBD5FF","#00A087FF","#3C5488FF")) +
  theme_publish() +
   theme(
    axis.title.x = element_text(size=16),
    axis.text.x  = element_text(size=16),
    axis.title.y = element_text(size=16),
    axis.text.y  = element_text(size=16),
    legend.title = element_text(size=16),
    legend.text = element_text(size=16),
    legend.position ="none") +
   ggtitle("Potential denitrification rate NO2 + N2)")

p15<-ggplot(dnf_data, aes(x=Conc, y=NO2sed, fill= Conc)) + 
  geom_dotplot(binaxis="y", stackdir="center") +
  geom_boxplot() +
  ylab(expression(nmol~N-NO[2]^{"-"}~~g^{-1}~h^{-1})) +
  xlab(expression(Cu~NPs~(µg~g^{-1}~sediment))) +
  scale_fill_manual(values = c("#E64B35FF","#4DBBD5FF","#00A087FF","#3C5488FF")) +
  theme_publish() +
   theme(
    axis.title.x = element_text(size=16),
    axis.text.x  = element_text(size=16),
    axis.title.y = element_text(size=16),
    axis.text.y  = element_text(size=16),
    legend.title = element_text(size=16),
    legend.text = element_text(size=16),
    legend.position ="none") +
   ggtitle("N-NO2 consumption rate")

plot_grid6 <- plot_grid(p14,p15,
          ncol=2, nrow=1)
plot_grid6

# Are these differences significant?

kruskal.test(DNF ~ Conc, data = dnf_data) 
kruskal.test(NO2sed ~ Conc, data = dnf_data) 
```
